aiida_ase/calculations/ase.py

```diff
@@ -278,7 +282,12 @@ def return_a_function(v):
         calcinfo.remote_copy_list = remote_copy_list
 
         codeinfo = common.CodeInfo()
-        codeinfo.cmdline_params = [self._INPUT_FILE_NAME]
+        cmdline_params = settings.pop('CMDLINE', [])
+        if not isinstance(cmdline_params, (list, tuple)):
+            cmdline_params = list(cmdline_params)
```

Comment:
I think we should be more strict here and only accept a list and otherwise raise. Imagine the user defines a dictionary, it will convert the dict to a list and try to append to a list, which might get weird behavior. I would suggest the following instead:

```suggestion
        cmdline_params = settings.pop('CMDLINE', [])
        if not isinstance(cmdline_params, (list, tuple)):
            raise InputValidationError('the `CMDLINE` key in the `settings` input should be a list or tuple.')
```